src/components/SelectionListSingle/ListItem/SingleSelectListItem.tsx

```diff
@@ -0,0 +1,66 @@
+import React, {useCallback} from 'react';
+import Checkbox from '@components/Checkbox';
+import useThemeStyles from '@hooks/useThemeStyles';
+import RadioListItem from './RadioListItem';
+import type {ListItem, SingleSelectListItemProps} from './types';
+
+/**
+ * SingleSelectListItem mirrors the behavior of a default RadioListItem, but adds support
+ * for the new style of single selection lists.
+ */
+function SingleSelectListItem<TItem extends ListItem>({
+    item,
+    isFocused,
+    showTooltip,
+    isDisabled,
+    onSelectRow,
+    onDismissError,
+    shouldPreventEnterKeySubmit,
+    isMultilineSupported = false,
+    isAlternateTextMultilineSupported = false,
+    alternateTextNumberOfLines = 2,
+    onFocus,
+    shouldSyncFocus,
+    wrapperStyle,
+    titleStyles,
+}: SingleSelectListItemProps<TItem>) {
+    const styles = useThemeStyles();
+    const isSelected = item.isSelected;
```

Comment:
I think this variable is unnecessary here - it's used only once in the whole file. It would also simplify the memoization below to provide only two dependencies instead of three